tests/unit/cmd/sca/test_scan_prepush.py

```diff
@@ -0,0 +1,147 @@
+from pathlib import Path
+from typing import List, Optional
+
+import pytest
+from click.testing import CliRunner
+
+from ggshield.cmd.main import cli
+from ggshield.core.errors import ExitCode
+from ggshield.secret.repo import cd
+from tests.repository import Repository
+from tests.unit.conftest import my_vcr
+
+
+@pytest.mark.parametrize(
+    "scan_arg, cassette",
+    [
+        (None, "test_sca_scan_prepush_output_diff"),
+        ("--all", "test_sca_scan_prepush_output_all"),
+    ],
+)
+def test_sca_scan_prepush_output(
+    tmp_path: Path,
+    cli_fs_runner: CliRunner,
+    scan_arg: Optional[List[str]],
+    cassette: str,
+    pipfile_lock_with_vuln,
+) -> None:
+    """
+    GIVEN a remote repository and a local clone
+    WHEN executing the prepush command with or without the '--all' option
+    THEN the scan output format is correct
+    """
+    remote_repo = Repository.create(tmp_path / "remote", bare=True)
```

Comment:
nit : I get why there is a remote_repo, but I'm not convinced this is necessary.

Reply:
I don't understand what you suggest to modify

Reply:
In this test, since there is no push, having a local_repo is enough, and we don't need to clone it from a remote repo. This line is not really useful. 
I don't know if you should really remove it since it still makes the context of the test clearer.

Reply:
I would prefer to leave it as is for the context if it's ok with you